src/components/lunatic-components.tsx

```diff
@@ -1,26 +1,35 @@
 import {
 	Fragment,
+	useRef,
 	type PropsWithChildren,
 	type ReactElement,
 	type ReactNode,
-	useRef,
 } from 'react';
-import * as lunaticComponents from './index';
-import type { FilledLunaticComponentProps } from '../use-lunatic/commons/fill-components/fill-components';
 import { useAutoFocus } from '../hooks/use-auto-focus';
-import { hasComponentType } from '../use-lunatic/commons/component';
+import { hasComponentType, hasLabel } from '../use-lunatic/commons/component';
+import type { FilledLunaticComponentProps } from '../use-lunatic/commons/fill-components/fill-components';
+import * as lunaticComponents from './index';
 
 type Props<T extends Record<string, unknown>> = {
 	// List of components to display (coming from getComponents)
-	components: (FilledLunaticComponentProps | ReactElement)[];
+	components: (
+		| (FilledLunaticComponentProps & { colspan?: number; rowspan?: number })
+		| { label: ReactElement; colspan?: number; rowspan?: number }
+	)[];
 	// Key that trigger autofocus when it changes (pageTag)
 	autoFocusKey?: string;
 	// Returns the list of extra props to add to components
-	componentProps?: (component: FilledLunaticComponentProps) => T;
+	componentProps?: (
+		component: FilledLunaticComponentProps & {
+			colspan?: number;
+			rowspan?: number;
```

Comment:
This seems to be a little bit hacky, I think correct way is to add `colspan` and `rowspan` props directly in `ComponentTableType`

Reply:
Yes, it will be more clear